packages/material-ui/src/Typography/Typography.js

```diff
@@ -1,106 +1,47 @@
 import * as React from 'react';
 import PropTypes from 'prop-types';
-import clsx from 'clsx';
-import { useThemeVariants } from '@material-ui/styles';
-import withStyles from '../styles/withStyles';
-import capitalize from '../utils/capitalize';
+import experimentalStyled from '../styles/experimentalStyled';
 
-export const styles = (theme) => ({
-  /* Styles applied to the root element. */
-  root: {
-    margin: 0,
-  },
-  /* Styles applied to the root element if `variant="body2"`. */
-  body2: theme.typography.body2,
-  /* Styles applied to the root element if `variant="body1"`. */
-  body1: theme.typography.body1,
-  /* Styles applied to the root element if `variant="caption"`. */
-  caption: theme.typography.caption,
-  /* Styles applied to the root element if `variant="button"`. */
-  button: theme.typography.button,
-  /* Styles applied to the root element if `variant="h1"`. */
-  h1: theme.typography.h1,
-  /* Styles applied to the root element if `variant="h2"`. */
-  h2: theme.typography.h2,
-  /* Styles applied to the root element if `variant="h3"`. */
-  h3: theme.typography.h3,
-  /* Styles applied to the root element if `variant="h4"`. */
-  h4: theme.typography.h4,
-  /* Styles applied to the root element if `variant="h5"`. */
-  h5: theme.typography.h5,
-  /* Styles applied to the root element if `variant="h6"`. */
-  h6: theme.typography.h6,
-  /* Styles applied to the root element if `variant="subtitle1"`. */
-  subtitle1: theme.typography.subtitle1,
-  /* Styles applied to the root element if `variant="subtitle2"`. */
-  subtitle2: theme.typography.subtitle2,
-  /* Styles applied to the root element if `variant="overline"`. */
-  overline: theme.typography.overline,
-  /* Styles applied to the root element if `variant="inherit"`. */
-  inherit: {},
-  /* Styles applied to the root element if `align="left"`. */
-  alignLeft: {
-    textAlign: 'left',
-  },
-  /* Styles applied to the root element if `align="center"`. */
-  alignCenter: {
-    textAlign: 'center',
-  },
-  /* Styles applied to the root element if `align="right"`. */
-  alignRight: {
-    textAlign: 'right',
-  },
-  /* Styles applied to the root element if `align="justify"`. */
-  alignJustify: {
-    textAlign: 'justify',
-  },
-  /* Styles applied to the root element if `nowrap={true}`. */
-  noWrap: {
+const getTextColor = (color, palette) => {
+  if (color === 'textPrimary' || color === 'textSecondary') {
+    return palette.text[color.split('text').pop().toLowerCase()];
+  }
+
+  if (color === 'inherit' || color === 'initial') {
+    return color;
+  }
+
+  return palette[color].main;
+};
+
+export const TypographyRoot = experimentalStyled(
+  'span',
+  {},
+  { muiName: 'MuiTypography' },
+)((props) => ({
```

Comment:
I like how the styles are much more compact :)

Reply:
To be fair, most of the reduction comes from the missing inline documentation. Last time I asked, core members did seem to rely on them so I'm not sure the new compactness a good thing (can't find the poll in slack anymore). Personally, I'm happy with getting rid of them to simplify documentation but there are some unintended side-effects to this change.

Reply:
Agree, that's true too, but still I like how all variants are simply defined by one line:

```
...(props.styleProps.variant && props.theme.typography[props.styleProps.variant]),
```

at the same time allowing us to support dynamic variants out of the box :)